apps/src/p5lab/AnimationPicker/AnimationPickerBody.jsx

```diff
@@ -231,6 +231,10 @@ export default class AnimationPickerBody extends React.Component {
     // Display second "Done" button. Useful for mobile, where the original "done" button might not be on screen when
     // animation picker is loaded. 600 pixels is minimum height of the animation picker.
     const shouldDisplaySecondDoneButton = isMobileDevice();
+    const showingUploadButton =
+      !hideUploadOption &&
+      ((!searching && (!inCategory || isBackgroundsTab)) ||
+        results.length === 0);
```

Comment:
Nit: There's a lot of nested conditional logic here. Is there a way this could be expressed in a way that's easier to read?

Reply:
good point. I'll split this out a little and add comments.